contributor_docs/fes_reference_dev_notes.md

```diff
@@ -0,0 +1,456 @@
+```
+ _fesErrorMonitor
+     processStack
+       printFriendlyError
+     (if type of error is ReferenceError)
+       _handleMisspelling
+         computeEditDistance
+         _report
+       _report
+       printFriendlyStack
+     (if type of error is SyntaxError, TypeError, etc)
+       _report
+       printFriendlyStack
+```
+##### Syntax
+````JavaScript
+fesErrorMonitor(event)
+````
+##### Parameters
+```
+@param {*}  e     Error event
+```
+##### Location
+core/friendly_errors/fes_core.js
+
+### `fesCodeReader()`
+##### Examples
+Example of Redefining p5.js Reserved Constant
+````JavaScript
+function setup() {
+  //PI is a p5.js reserved constant
+  let PI = 100;
+}
+// FES will show:
+// > 🌸 p5.js says: you have used a p5.js reserved variable "PI" make sure you change the variable name to something else. (https://p5js.org/reference/#/p5/PI)
+````
+Example of Redefining p5.js Reserved Function
+````JavaScript
+function setup() {
+  //text is a p5.js reserved function
+  let text = 100;
+}
+// FES will show:
+// > 🌸 p5.js says: you have used a p5.js reserved function "text" make sure you change the function name to something else.
+````
+##### Description
+`fesCodeReader()` checks (1) if any p5.js constant or function is used outside of the `setup()` and/or `draw()` function and (2) if any p5.js reserved constant or function is redeclared.
+
+Generates and print a friendly error message with type: `fes.sketchReaderErrors.reservedConst`, `fes.sketchReaderErrors.reservedFunc`.
+
+In `setup()` and `draw()` functions it performs:
+ * Extraction of the code written by the user
+ * Conversion of the code to an array of lines of code
+ * Catching variable and function declaration
+ * Checking if the declared function/variable is a reserved p5.js constant or function and report it.
+
+This function is executed whenever the `load` event is triggered.
+
+##### Location
+core/friendly_errors/sketch_reader.js
+
+### `checkForUserDefinedFunctions()`
+##### Examples
+````JavaScript
+function preLoad() {
+  loadImage('myimage.png');
+}
+// FES will show:
+// > 🌸 p5.js says: It seems that you may have accidentally written preLoad instead of preload. Please correct it if it's not intentional. (http://p5js.org/reference/#/p5/preload)
+````
+##### Description
+Checks if any user defined function (`setup()`, `draw()`, `mouseMoved()`, etc.) has been used with a capitalization mistake.
+
+Generates and prints a friendly error message using key: `fes.checkUserDefinedFns`.
+##### Syntax
+````JavaScript
+checkForUserDefinedFunctions(context)
+````
+##### Parameters
+```
+@param {*} context  Current default context.
+                    Set to window in "global mode" and
+                    to a p5 instance in "instance mode"
+```
+##### Location
+core/friendly_errors/fes_core.js
+
+### `_friendlyAutoplayError()`
+##### Description
+`_friendlyAutoplayError()` is called internally if there is an error with autoplay.
```

Comment:
I don't know what autoplay is 😅 maybe a short description of what this refers to would be helpful?

Reply:
Good catch! Will add a description here 😄